.github/workflows/deps.yaml

```diff
@@ -167,7 +179,121 @@ jobs:
         with:
           name: linux_arm64_substrait
           path: deps/linux_arm64
+      - uses: actions/download-artifact@v4
+        with:
+          name: iceberg-v1.1.3-extension-linux_arm64
+          path: deps/linux_arm64
       - name: Push static libraries
         uses: stefanzweifel/git-auto-commit-action@v5
         with:
           commit_message: Re-build static libraries
+
+  iceberg:
+    name: Build extension binaries
+    uses: ./.github/workflows/iceberg_extension_distribution.yaml
+    with:
+      extension_name: iceberg
+      duckdb_version: v1.1.3
+      ci_tools_version: v1.1.3
+      exclude_archs: 'linux_amd64_gcc4'
+
+  test_darwin_amd64:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install avrocpp
+        shell: bash
+        run: |
+          brew install avro-cpp
+
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.23'
+
+      - name: Test loading the extensions
+        run: go test -run ^TestLoaded
+
+  test_darwin_arm64:
+    runs-on: macos-13
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install avrocpp
+        shell: bash
+        run: |
+          brew install avro-cpp
```

Comment:
how do we control the avro version here? I wonder if we should build from source here too?

Reply:
Compiled for macos as well.